pkg/plugin/oauth2/in_memory_repository.go

```diff
@@ -51,6 +51,20 @@ func (r *InMemoryRepository) FindByTokenURL(url url.URL) (*OAuth, error) {
 	return nil, ErrOauthServerNotFound
 }
 
+// Persist add a new OAuth Server to the repository
+func (r *InMemoryRepository) Persist(server *OAuth) error {
```

Comment:
What is the difference between `Persist` and `Add`?

Reply:
Add uses `mgo.Upsert` and never fails in case of duplicate constraints. Persist relies on `mgo.Insert` method and run constraint validations

Reply:
In fact I'd rather to have:
```go
Add () {
  mgo.Insert
}

Save () {
  mgo.Upsert
}
```

But it would require some refactoring

Reply:
I like All/Save approach